### PR TITLE
[DDW-1222] Catalyst UI improvements

### PR DESCRIPTION
**Pull request by marcin-mazurek**
This PR improves the Catalyst voting UI by removing confusing next fund snapshot date (we will only show info around the current fund) as well as removes any mentions of the fund number from the registration flow (since the user registers for Catalyst voting in general and not for a specific fund).

## Screenshots

Screenshot 2023-07-06 at 23 19 46
Screenshot 2023-07-06 at 23 19 55
Screenshot 2023-07-06 at 23 20 04
Screenshot 2023-07-06 at 23 22 23
Screenshot 2023-07-06 at 23 22 32
Screenshot 2023-07-06 at 23 20 10
Screenshot 2023-07-06 at 23 26 08

## Testing Checklist

- [Slack QA thread](https://input-output-rnd.slack.com/messages/GGKFXSKC6)
- [ ] Test

---

### Basics

- [x] PR assigned to the PR author(s)
- [x] `input-output-hk/daedalus-dev` and `input-output-hk/daedalus-qa` assigned as PR reviewers
- [x] PR has appropriate labels (`release-vNext`, `feature`/`bug`/`chore`, `WIP`)
- [x] PR link is added to a Jira ticket, ticket moved to In Review
- [x] PR is updated to the most recent version of the target branch (and there are no conflicts)
- [x] PR has a good description that summarizes all changes
- [x] PR contains screenshots (in case of UI changes)
- [x] CHANGELOG entry has been added to the top of the appropriate section (_Features_, _Fixes_, _Chores_) and is linked to the correct PR on GitHub
- [x] There are no missing translations (running `yarn manage:translations` produces no changes)
- [x] Text changes are proofread and approved (Jane Wild / Amy Reeve)
- [x] Japanese text changes are proofread and approved (Junko Oda)
- [x] Storybook works and no stories are broken (`yarn storybook`)
- [x] In case of dependency changes `yarn.lock` file is updated

### Code Quality

- [ ] Important parts of the code are properly commented and documented
- [ ] Code is properly typed with typescript types
- [ ] React components are split-up enough to avoid unnecessary re-renderings
- [ ] Any code that only works in main process is neatly separated from components

### Testing

- [ ] New feature/change is covered by acceptance tests
- [ ] New feature/change is manually tested and approved by QA team
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature/change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review

- [ ] Update Slack QA thread by marking it with a green checkmark

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 ### Features
 
-- Show "To be defined" if date of the next Catalyst fund is in the past ([PR 3105](https://github.com/input-output-hk/daedalus/pull/3105))
```
Comment (marcin-mazurek) [inline CHANGELOG.md:7]: Replacing CHANGELOG entry on purpose - effectively, what was delivered in this unreleased PR, was deleted

File: source/renderer/app/components/voting/voting-registration-wizard-steps/VotingRegistrationStepsEnterPinCode.tsx [hunk 2]
```diff
@@ -162,15 +161,9 @@ class VotingRegistrationStepsEnterPinCode extends Component<Props> {
...
-        nextFundNumber={nextFundNumber}
       >
         <p className={styles.description}>
-          <FormattedHTMLMessage
-            {...messages.description}
-            values={{
-              nextVotingFundNumber: nextFundNumber,
-            }}
-          />
+          <FormattedHTMLMessage {...messages.description} />
```
Comment (tomislavhoracek) [inline source/renderer/app/components/voting/voting-registration-wizard-steps/VotingRegistrationStepsEnterPinCode.tsx:166]: Can we use just `intl.formatMessage(messages.description)` here?

File: source/renderer/app/components/voting/voting-registration-wizard-steps/VotingRegistrationStepsRegister.tsx [hunk 3]
```diff
@@ -186,15 +184,9 @@ class VotingRegistrationStepsRegister extends Component<Props> {
...
-        nextFundNumber={nextFundNumber}
       >
         <p className={styles.description}>
-          <FormattedHTMLMessage
-            {...messages.description}
-            values={{
-              nextVotingFundNumber: nextFundNumber,
-            }}
-          />
+          <FormattedHTMLMessage {...messages.description} />
```
Comment (tomislavhoracek) [inline source/renderer/app/components/voting/voting-registration-wizard-steps/VotingRegistrationStepsRegister.tsx:189]: Same as previous comment, `intl.formatMessage(messages.description);` ?

File: storybook/stories/voting/Voting.stories.tsx [hunk 4]
```diff
@@ -174,17 +171,15 @@ storiesOf('Voting / Voting Registration Wizard', module)
...
-      nextFundNumber={8}
     />
   ))
   .add('Voting Registration - Step 5', () => (
     <VotingRegistrationStepsQrCode
       onClose={action('onClose')}
       onDownloadPDF={action('onDownloadPDF')}
       stepsList={stepsList}
-      activeStep={2}
+      activeStep={5}
```
Comment (tomislavhoracek) [inline storybook/stories/voting/Voting.stories.tsx:181]: Nice catch 👌

File: source/renderer/app/components/voting/voting-registration-wizard-steps/VotingRegistrationStepsEnterPinCode.tsx (hunk 2 above)
Comment (marcin-mazurek) [reply source/renderer/app/components/voting/voting-registration-wizard-steps/VotingRegistrationStepsEnterPinCode.tsx]: Resolved in c076a3917e896a89cfb497d66c21828eacb5ecd3, thanks!

File: source/renderer/app/components/voting/voting-registration-wizard-steps/VotingRegistrationStepsRegister.tsx (hunk 3 above)
Comment (marcin-mazurek) [reply source/renderer/app/components/voting/voting-registration-wizard-steps/VotingRegistrationStepsRegister.tsx]: Resolved in c076a3917e896a89cfb497d66c21828eacb5ecd3, good catch!

File: storybook/stories/voting/Voting.stories.tsx (hunk 4 above)
Comment (marcin-mazurek) [reply storybook/stories/voting/Voting.stories.tsx:181]: Thanks!

**Review by tomislavhoracek — APPROVED** [review]
@marcin-mazurek nice job 💯

**Comment by marcin-mazurek** [comment]
Credits to @lucas-barros too ;)

**Review by DominikGuzei — APPROVED** [review]
much better like this 👏

**Comment by gabriela-ponce** [comment]
@marcin-mazurek this looks ok overall, but there is a detail with the default name of the pdf file on the last step.
Previously it was _["fundXX-voting-registration-WalletName-Date.pdf"](https://drive.google.com/file/d/1ZIse4KRfXKoGxA6Y41DsXOQyDm1lfiQe/view?usp=drive_link)_.  On the latest build the name is _["fund-voting-registration-WalletName-Date.pdf"](https://drive.google.com/file/d/1t0uqTkIc8RnsCp1tyYGiucE6pzYbFzCW/view?usp=drive_link)_. Maybe the word "fund" could be removed? It seems irrelevant now.

**Comment by marcin-mazurek** [comment]
@gabriela-ponce good point, I've replaced "fund" with "catalyst" - can you re-check?
image

**Comment by miorsufianiohk** [comment]
Hi @marcin-mazurek 
The PR is looking good so far. However, I found an issue which is not related to this PR which I could reproduce on production and dev, The `Newsletter `link appears to be broken. So, I have been wondering   if we are going to fix this at all? or whether in the scope of this PR or another PR? [See screenshot](https://drive.google.com/file/d/1UqzhJ1KLcQuTxrzhlUJ2dhSevH6O-xKl/view?usp=sharing)

**Comment by marcin-mazurek** [comment]
@miorsufianiohk I asked Catalyst team to review all links, seems like there is more links that require updating. Good catch!

**Review by miorsufianiohk — APPROVED** [review]
LGTM. Great work @marcin-mazurek 👍

**Review by gabriela-ponce — APPROVED** [review]

**Pull request merged**